WatchFaceAlphaKotlin/app/src/main/java/com/example/android/wearable/alpha/editor/WatchFaceConfigActivity.kt

```diff
@@ -1,17 +1,137 @@
+    private val stateHolder: WatchFaceConfigStateHolder by lazy {
+        WatchFaceConfigStateHolder(
+            lifecycleScope,
+            this@WatchFaceConfigActivity,
+            intent
+        )
+    }
+
+    private lateinit var binding: ActivityWatchFaceConfigBinding
+
+    private var initialDataLoaded = false
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_watch_face_config)
         Log.d(TAG, "onCreate()")
+
+        binding = ActivityWatchFaceConfigBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // Disable widgets until data loads and values are set.
+        binding.colorStylePickerButton.isEnabled = false
+        binding.ticksEnabledSwitch.isEnabled = false
+        binding.minuteHandLengthSlider.isEnabled = false
+
+        // Set max and min.
+        binding.minuteHandLengthSlider.valueTo = MINUTE_HAND_LENGTH_MAXIMUM_FOR_SLIDER
+        binding.minuteHandLengthSlider.valueFrom = MINUTE_HAND_LENGTH_MINIMUM_FOR_SLIDER
+        binding.minuteHandLengthSlider.value = MINUTE_HAND_LENGTH_DEFAULT_FOR_SLIDER
+
+        binding.minuteHandLengthSlider.addOnChangeListener { slider, value, fromUser ->
+            Log.d(TAG, "addOnChangeListener(): $slider, $value, $fromUser")
+            stateHolder.setMinuteHandArmLength(value)
+        }
+
+        lifecycleScope.launch(Dispatchers.Main.immediate) {
+            stateHolder.uiState.collect { uiState: WatchFaceConfigStateHolder.EditWatchFaceUiState ->
+                when (uiState) {
+                    is WatchFaceConfigStateHolder.EditWatchFaceUiState.Loading -> {
+                        Log.d(TAG, "StateFlow Loading: ${uiState.message}")
+                    }
+                    is WatchFaceConfigStateHolder.EditWatchFaceUiState.Success -> {
+                        Log.d(TAG, "StateFlow Success.")
+                        updateWatchFacePreview(uiState.userStylesAndPreview)
+                    }
+                    is WatchFaceConfigStateHolder.EditWatchFaceUiState.Error -> {
+                        Log.e(TAG, "Flow error: ${uiState.exception}")
+                    }
+                }
+            }
+        }
+    }
+
+    private fun updateWatchFacePreview(
+        userStylesAndPreview: WatchFaceConfigStateHolder.UserStylesAndPreview
+    ) {
+        Log.d(TAG, "updateWatchFacePreview: $userStylesAndPreview")
+
+        val colorStyleId: String = userStylesAndPreview.colorStyleId
+        Log.d(TAG, "\tselected color style: $colorStyleId")
+
+        binding.ticksEnabledSwitch.isChecked = userStylesAndPreview.ticksEnabled
+        binding.minuteHandLengthSlider.value = userStylesAndPreview.minuteHandLength
+        binding.preview.watchFaceBackground.setImageBitmap(userStylesAndPreview.previewImage)
+
+        if (!initialDataLoaded) {
+            initialDataLoaded = true
+            enabledWidgets()
+        }
```

Comment:
Nit: just a small personal preference, but repeatedly setting `isEnabled=true` here doesn't have any downside, so why not just always set them? We avoid needing a flag, and makes our state handling overall more idempotent.

Reply:
I'm confused, it's only set to true once and never done again, i.e., it's false by default and so this block will only be executed when it is false. After that, it sets it to true, enables the widgets and is never called again.

Reply:
Right, I was think we could just do this:

```suggestion
        enabledWidgets()
```

Doesn't change any behavior, but reduces the number of branches to worry about.